services/s3/src/main/java/software/amazon/awssdk/services/s3/internal/crt/S3NativeClientConfiguration.java

```diff
@@ -65,6 +67,8 @@ public S3NativeClientConfiguration(Builder builder) {
         this.endpointOverride = builder.endpointOverride;
 
         this.checksumValidationEnabled = builder.checksumValidationEnabled == null || builder.checksumValidationEnabled;
+        this.readBufferSizeInBytes = builder.readBufferSizeInBytes == null ?
```

Comment:
What happens if user explicitly passes 
1. `readBufferSizeInBytes = 0 ' 
2.  negative value 
3.  " readBufferSizeInBytes < partSizeInBytes * 10 " 
in this case should we assign default as  partSizeInBytes * 10  ?

Reply:
Good call, I'll add validation. If a valid buffer size is provided, we will just use that one